doc/guidelines/frontend-testing.md

```diff
@@ -0,0 +1,81 @@
+# Frontend Testing Guidelines: Writing Resilient E2E Tests
+
+These guidelines will help you write frontend code that enables resilient and user-centric test execution.
+They are mostly inspired by Playwrights Best Practices.
+When following the points below, you will be on a good way to write good testable code.
+
+## 1. Locators
+
+In order to find elements on the webpage, tests will need to use [locators](https://playwright.dev/docs/locators):
```

Comment:
The main readers of such guidelines are PermaplanT developers, who also write tests. So IMHO we should have guidelines which focus on explaining how to write testable html code. Or is this topic not so important anymore and tests will introduce roles etc. everywhere themselves?

This doesn't mean that this guideline should be rewritten, tester-focused guidelines are also important but somehow it doesn't match the title of this PR "Frontend coding guidelines".

Reply:
> Or is this topic not so important anymore and tests will introduce roles etc. everywhere themselves?

Hopefully it wont be only testers writing locators. This should only happen as a last resort.

> somehow it doesn't match the title of this PR "Frontend coding guidelines".

I was not really sure back then what I m really going to write about here.
Now I renamed it to focus only on locators.

Reply:
> Hopefully it wont be only testers writing locators. This should only happen as a last resort.

It should be clearly said who is responsible for adding `data-testid`. Probably it actually should be only by testers?